src/app/test_executive/test_executive.ml

```diff
@@ -68,53 +68,54 @@ let main inputs =
-      in
-      let%bind log_engine_error_set =
-        Option.value_map !log_engine_ref
-          ~default:(Deferred.return Test_error.Set.empty) ~f:(fun log_engine ->
-            match%map Engine.Log_engine.destroy log_engine with
-            | Ok errors ->
-                errors
-            | Error err ->
-                Test_error.Set.hard_singleton (Test_error.internal_error err)
-        )
-      in
-      let%bind () =
-        Option.value_map !net_manager_ref ~default:Deferred.unit
-          ~f:Engine.Network_manager.cleanup
+    let cleanup () =
+      let cleanup_result =
+        let open Malleable_error.Let_syntax in
+        (* let%bind () = test_error in *)
+        let%bind remote_error_set =
+          Option.value_map !log_engine_ref
+            ~default:(Malleable_error.return Test_error.Set.empty)
+            ~f:Engine.Log_engine.destroy
+        in
+        let%map () =
+          Option.value_map !net_manager_ref ~default:Malleable_error.ok_unit
+            ~f:
+              (Fn.compose
+                 (Deferred.bind ~f:Malleable_error.return)
+                 Engine.Network_manager.cleanup)
+        in
+        remote_error_set
       in
-      report_test_errors
-        (Test_error.Set.combine [test_error_set; log_engine_error_set])
+      match%bind Malleable_error.lift_error_set cleanup_result with
+      | Ok (remote_error_set, internal_error_set) ->
+          report_test_errors
+            (Test_error.Set.combine [remote_error_set; internal_error_set])
+      | Error internal_error_set ->
+          report_test_errors internal_error_set
+    in
+    let%bind test_error_str =
+      match%map test_error with
+      | Ok {Malleable_error.Accumulator.computation_result= _; soft_errors= _}
+        ->
+          "<none>"
+      | Error
+          { Malleable_error.Hard_fail.hard_error= hard_err
+          ; Malleable_error.Hard_fail.soft_errors= _ } ->
+          Error.to_string_hum hard_err.error
     in
     match !cleanup_deferred_ref with
     | Some deferred ->
         [%log error]
           "additional call to cleanup testnet while already cleaning up \
-           ($reason, $error)"
+           ($reason, $hard_error)"
           ~metadata:
-            [ ("reason", `String reason)
-            ; ( "error"
-              , `String
-                  ( Option.map test_error ~f:Error.to_string_hum
-                  |> Option.value ~default:"<none>" ) ) ] ;
+            [("reason", `String reason); ("hard_error", `String test_error_str)] ;
         deferred
     | None ->
         [%log info] "cleaning up testnet ($reason, $error)"
           ~metadata:
-            [ ("reason", `String reason)
-            ; ( "error"
-              , `String
-                  ( Option.map test_error ~f:Error.to_string_hum
-                  |> Option.value ~default:"<none>" ) ) ] ;
-        let deferred = cleanup test_error in
+            [("reason", `String reason); ("error", `String test_error_str)] ;
```

Comment:
It's better to let the metadata keys for the log statements in each of these branches match. Makes it easier to write stackdriver and logproc filters later when you want to inspect the logs. The log on 108 calls the error field `hard_error`, and the one here on 117 calls it `error`.